luigi/contrib/k8s_job.py

```diff
@@ -0,0 +1,193 @@
+
+        For more details please referer to:
+        http://kubernetes.io/docs/user-guide/kubeconfig-file
+        """
+        return configuration.get_config().get("k8s", "kubeconfig_path", "~/.kube/config")
+
+    @property
+    def name(self):
+        """
+        A name for this job. This task will automatically append a UUID to the
+        name before to submit to Kubernetes.
+        """
+        pass
+
+    @property
+    def spec_schema(self):
+        """
+        Kubernetes Job spec schema in JSON format, example::
+
+            {
+                "containers": [{
+                    "name": "pi",
+                    "image": "perl",
+                    "command": ["perl",  "-Mbignum=bpi", "-wle", "print bpi(2000)"]
+                }],
+                "restartPolicy": "Never"
+            }
+
+        **restartPolicy**
+
+        - If restartPolicy is not defined, it will be set to "Never" by default.
+
+        - **Warning**: restartPolicy=OnFailure will bypass max_retrials, and restart
+        the container until success, with the risk of blocking the Luigi task.
+
+        For more informations please refer to:
+        http://kubernetes.io/docs/user-guide/pods/multi-container/#the-spec-schema
+        """
+
+    @property
+    def max_retrials(self):
+        """
+        Maximum number of retrials in case of failure.
+        """
+        return 0
+
+    def __track_job(self):
+        """Poll job status while active"""
+        while (self.__get_job_status() == "running"):
+            self.__logger.debug("Kubernetes job " + self.uu_name + " is still running")
+            time.sleep(self.__POLL_TIME)
+        if(self.__get_job_status() == "succeeded"):
+            self.__logger.info("Kubernetes job " + self.uu_name + " succeeded")
+        else:
+            raise RuntimeError("Kubernetes job " + self.uu_name + " failed")
+
+    def __get_job_status(self):
+        """It returns the Kubernetes job status"""
+        # Look for the required job
+        jobs = Job.objects(self.__kube_api).filter(selector="luigi_task_id=" + self.job_uuid)
+        # Raise an exception if no such job found
+        if len(jobs.response["items"]) == 0:
+            raise RuntimeError("Kubernetes job " + self.uu_name + " not found")
+        # Figure out status and return it
+        job = Job(self.__kube_api, jobs.response["items"][0])
+        if ("succeeded" in job.obj["status"] and job.obj["status"]["succeeded"] > 0):
+            job.scale(replicas=0) # Downscale the job, but keep it there for logging
+            return "succeeded"
+        if ("failed" in job.obj["status"]):
+            failed_cnt = job.obj["status"]["failed"]
+            self.__logger.debug("Kubernetes job " + self.uu_name +
+                " status.failed: " + str(failed_cnt))
+            if (failed_cnt > self.max_retrials):
+                job.scale(replicas=0) # avoid more retrials
+                return "failed"
+        return "running"
+
```

Comment:
I think adding this to the method would be really helpful, and fixes an error that I had while testing (there's no way to tell that the k8s job is complete)

```python
def output(self):
        """Implement an output to allow for dependency chaining"""
        raise NotImplementedError("Subclass must define output")
```